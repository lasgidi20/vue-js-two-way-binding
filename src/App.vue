<template>
  <ChildComponent @bool="bool" :text="text"/>
</template>

<script setup>
  import ChildComponent from './components/ChildComponent.vue';
  import {ref} from 'vue';
  const bool = ref(true)
  const text = ref("Mike")
</script>
<style scoped>
</style>