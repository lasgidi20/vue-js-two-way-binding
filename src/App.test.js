import { mount } from '@vue/test-utils'
import App from './../src/App.vue'
describe('App Component', () =>  {
  const wrapper = mount(App)
  test('renders the App', () => {
    expect(wrapper.text()).toContain('Mike')
    expect(wrapper.text()).toContain('change bool')
    expect(wrapper.text()).toMatchSnapshot()
  })

  test("has a button for changing boolean", () => {
    expect(wrapper.find("button").exists()).toBe(true);
  });

  test("has an input field", () => {
    expect(wrapper.find("input").exists()).toBe(true);
  });

  test("sets input field", async () => {
    await wrapper.get('[data-test="dynamic-field"]').setValue("XFACTOR");
    expect(wrapper.findAll('[data-test="dynamic-field"]')).toHaveLength(1);
    expect(wrapper.text()).toContain('XFACTOR')
    await wrapper.get('[data-test="dynamic-field"]').setValue("New Data");
    expect(wrapper.text()).toContain('New Data')
  });

  test("updates boolean variable with button click", async () => {
     await wrapper.find("button").trigger('click')
     expect(wrapper.text()).toContain("bool: false")
     await wrapper.find("button").trigger('click')
     expect(wrapper.text()).toContain("bool: true")
  });
})