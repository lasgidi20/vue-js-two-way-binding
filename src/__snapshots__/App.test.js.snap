// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > renders the App 1`] = `"Parent: bool: true text: MikeChild:bool: truetext: Mikechange bool"`;

exports[`App Component > updates boolean variable with button 1`] = `"Parent: bool: true text: XFACTORChild:bool: truetext: XFACTORchange bool"`;
